src/AppBundle/Validator/Constraints/RecaptchaValidator.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/*
+ * This file is part of the Sententiaregum project.
+ *
+ * (c) Maximilian Bosch <maximilian.bosch.27@gmail.com>
+ * (c) Ben Bieler <benjaminbieler2014@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace AppBundle\Validator\Constraints;
+
+use ReCaptcha\ReCaptcha;
+use Symfony\Component\DependencyInjection\Container;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+/**
+ * Class RecaptchaValidator.
+ *
+ * @author Benjamin Bieler <benjaminbieler2014@gmail.com>
+ */
+class RecaptchaValidator extends ConstraintValidator
+{
+    /**
+     * Constructor.
+     *
+     * @param Container $container
+     */
+    public function __construct(Container $container)
+    {
+        $this->container = $container;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function validate($value, Constraint $constraint)
+    {
+
+        if (!$constraint instanceof Recaptcha) {
+            throw new UnexpectedTypeException($constraint, Recaptcha::class);
+        }
+
+        $recaptcha = new ReCaptcha($this->container->getParameter('secret_key'));
+
+        $check = $recaptcha->verify($constraint->recaptchaHash, $this->container->getParameter('site_url'));
+
+        if (!$check->isSuccess()) {
+            $this->context->buildViolation($constraint->message)
+                ->setInvalidValue($value)
+                ->atPath($constraint->propertyPath)
```

Comment:
thought about it again, remove `atPath` and keep `setInvalidValue`. My previous comment that `atPath` is necessary here too was garbage, sry.